module/Company/src/Company/Service/Company.php

```diff
@@ -1,39 +1,204 @@
 <?php
 
 namespace Company\Service;
-
-use Application\Service\AbstractService;
+//use Application\Service\AbstractService;
+use Application\Service\AbstractAclService;
 
 use Company\Model\Company as CompanyModel;
 use Company\Mapper\Company as CompanyMapper;
 /**
  * Company service.
  */
-class Company extends AbstractService
+class Company extends AbstractACLService
 {
     public function getCompanyList()
     {
-        return $this->getCompanyMapper()->findAll();
+        $translator = $this->getTranslator();
+        if ($this->isAllowed('list')){
+
+            return $this->getCompanyMapper()->findAll($translator->getLocale());
+        } else {
+            throw new \User\Permissions\NotAllowedException(
+                $translator->translate('You are not allowed list the companies')
+            );
+        }
     }
-    public function getCompaniesWithAsciiName($asciiName)
+    // Company list for admin interface
+    public function getHiddenCompanyList() 
     {
-        return $this->getCompanyMapper()->findCompaniesWithAsciiName($asciiName);
+        if ($this->isAllowed('listall')){
+
+            return $this->getCompanyMapper()->findAll();
+        } else {
+            $translator = $this->getTranslator();
+            throw new \User\Permissions\NotAllowedException(
+                $translator->translate('You are not allowed to acces the admin interface')
+            );
+        }
+    }
+    
+    public function saveCompany()
+    {
+        $this->getCompanyMapper()->save();
     }
 
-    public function getJobsWithAsciiName($companyAsciiName,$jobAsciiName)
+    public function savePacket()
     {
-        return $this->getJobMapper()->findJobWithAsciiName($companyAsciiName,$jobAsciiName);
+        $this->getPacketMapper()->save();
     }
-    public function getCompanyMapper()
+
+    public function insertCompany($languages)
     {
-        return $this->sm->get('company_mapper_company');
+        if ($this->isAllowed('insert')){
+            return $this->getCompanyMapper()->insert($languages);
+        }
+        else{
+            $translator = $this->getTranslator();
+            throw new \User\Permissions\NotAllowedException(
+                $translator->translate('You are not allowed to insert a company')
+            );
+        }
+
+    }
+        
+    public function insertPacketForCompanySlugName($companySlugName)
+    {
+        if ($this->isAllowed('insert')){
+            $companies = $this->getEditableCompaniesWithSlugName($companySlugName);
+            var_dump($companySlugName);
+            $company = $companies[0];
+            return $this->getPacketMapper()->insertPacketIntoCompany($company);
+        }
+        else{
```

Comment:
Don't use `else` if you don't need to!

In general it is good to avoid using `else` in classes. See http://www.slideshare.net/guilhermeblanco/object-calisthenics-applied-to-php